app.rb

```diff
@@ -1,5 +1,36 @@
 require 'sinatra/base'
+
 class RockPaperScissors < Sinatra::Base
+
+enable :sessions
+
+  get '/' do
+    erb :first
+  end
+
+  post '/names' do
+    session['name'] = params[:name]
+    redirect '/play'
+  end
+
+  get '/play' do
+    @name = session['name']
+    erb :play
+  end
+
+  post '/result' do
```

Comment:
Using sessions here is the right idea for single responsibility etc, but you can now extract the bottom 4 lines into a separate get block, and add a redirect in this post block to the new get block. That way your post is only taking in the params and putting them in the session. Then your get block is taking the params from the session and using them to render a page.